### DISCUSSION TITLE
Confusion in function object and function object map

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3465] [Q&A] [answered]
From the information provided in the documentation/user guide, I assumed the function object f, like in the given example can be evaluated at a single point and map() is used to evaluate the function at multiple points instead of a for loop. But even f can be evaluated at mulitple points by passing snapshots of the input as shown in the example. Are both equivalent ( ``` f(cs.pi*cs.DM.ones(1,10),cs.pi*cs.DM.ones(1,10))``` and ```F(cs.pi*cs.DM.ones(1,10),cs.pi*cs.DM.ones(1,10)) ```)?
 
```
x=cs.SX.sym('x',1)
y=cs.SX.sym('y',1)
f = cs.Function('f',[x,y],[cs.vertcat(cs.sin(x**5)**2+cs.cos(y**2)**2,cs.sin(x)**2+cs.cos(y)**2)])
F=f.map(10)
print(f)
print(F)
print(f(cs.pi*cs.DM.ones(1,10),cs.pi*cs.DM.ones(1,10)))
print(cs.n_nodes(f(cs.repmat(x,1,10),cs.repmat(x,1,10))))
print(F(cs.pi*cs.DM.ones(1,10),cs.pi*cs.DM.ones(1,10)))
print(cs.n_nodes(F(cs.repmat(x,1,10),cs.repmat(x,1,10))))
```
```
f:(i0,i1)->(o0[2]) SXFunction
map10_f:(i0[1x10],i1[1x10])->(o0[2x10]) Map

[[1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547], 
 [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]
47

[[1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547, 1.73547], 
 [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]
47

```

**Comment by jgillis** [comment, marked as answer]
CasADi Functions do support horizontal broadcasting.
Broacasting is equivalent to `map(N,'unroll')`.
For MX, `map(N,'unroll')` is more memory intense than `map(N)` (default is 'serial') and slower than `map(N,'thread')`.
For SX, there is a full equivalence, since everything gets inline.

The functionalities are related, but may dramatically differ in effect.
